frontend/src/scenes/data-warehouse/editor/QueryWindow.tsx

```diff
@@ -1,27 +1,110 @@
 import { Monaco } from '@monaco-editor/react'
-import { useActions, useValues } from 'kea'
+import { BindLogic, useActions, useValues } from 'kea'
 import { router } from 'kea-router'
 import type { editor as importedEditor } from 'monaco-editor'
 import { useState } from 'react'
 
+import { dataNodeLogic, DataNodeLogicProps } from '~/queries/nodes/DataNode/dataNodeLogic'
+import { variableModalLogic } from '~/queries/nodes/DataVisualization/Components/Variables/variableModalLogic'
+import {
+    variablesLogic,
+    VariablesLogicProps,
+} from '~/queries/nodes/DataVisualization/Components/Variables/variablesLogic'
+import {
+    dataVisualizationLogic,
+    DataVisualizationLogicProps,
+} from '~/queries/nodes/DataVisualization/dataVisualizationLogic'
+import { displayLogic } from '~/queries/nodes/DataVisualization/displayLogic'
+import { DataVisualizationNode, NodeKind } from '~/queries/schema'
+import { ItemMode } from '~/types'
+
 import { multitabEditorLogic } from './multitabEditorLogic'
 import { OutputPane } from './OutputPane'
 import { QueryPane } from './QueryPane'
 import { QueryTabs } from './QueryTabs'
 
 export function QueryWindow(): JSX.Element {
+    const [querySource, localSetQuerySource] = useState({
+        kind: NodeKind.DataVisualizationNode,
+        source: {
+            kind: NodeKind.HogQLQuery,
+            query: '',
+        },
+    } as DataVisualizationNode)
+
+    const vizKey = `SQLEditorScene`
+
+    const dataVisualizationLogicProps: DataVisualizationLogicProps = {
+        key: vizKey,
+        query: querySource,
+        dashboardId: undefined,
+        dataNodeCollectionId: vizKey,
+        insightMode: ItemMode.Edit,
+        loadPriority: undefined,
+        cachedResults: undefined,
+        variablesOverride: undefined,
+        setQuery: localSetQuerySource,
+    }
+
+    const dataNodeLogicProps: DataNodeLogicProps = {
+        query: querySource.source,
+        key: vizKey,
+        cachedResults: undefined,
+        loadPriority: undefined,
+        dataNodeCollectionId: vizKey,
+        variablesOverride: undefined,
+    }
+
+    const variablesLogicProps: VariablesLogicProps = {
+        key: dataVisualizationLogicProps.key,
+        readOnly: false,
+    }
+
+    return (
+        <BindLogic logic={dataNodeLogic} props={dataNodeLogicProps}>
```

Comment:
These logics need some serious rethinking. I have a better understanding of the relationship of these viz logics but still need to figure out how we can redo this without disrupting the rest of the app